_posts/2019-06-07-python-datascience-with-openfaas.md

```diff
@@ -0,0 +1,275 @@
+---
+title: Serving Data Science Models with OpenFaaS
+description: "Lucas will build and deploy a name classification function using OpenFaaS and PyTorch"
+date: 2019-06-01
+image: /images/pydatascience-template/artificial-neural-network-3501528_640.png
+categories:
+  - python
+  - coding
+  - examples
+  - datascience
+  - templates
+author_staff_member: lucas
+dark_background: false
+---
+
+Last month the OpenFaaS team was at [Kubecon Barcelona][kubecon-homepage], a frequent topic was deploying data science models to Kubernetes and, of course, can OpenFaaS help deploy models? In this post we will introduce a new function template aimed at Python data scientists and walk through a concrete example of deploying a [PyTorch][pytorch-homepage] model.
+
+<blockquote class="twitter-tweet"><p lang="en" dir="ltr">Up bright and early? We&#39;re meeting for an OpenFaaS f2f with end-users, contributors and community. 10.20am outside the keynote hall. See you there? <a href="https://twitter.com/hashtag/serverless?src=hash&amp;ref_src=twsrc%5Etfw">#serverless</a> <a href="https://twitter.com/hashtag/community?src=hash&amp;ref_src=twsrc%5Etfw">#community</a> <a href="https://twitter.com/hashtag/faas?src=hash&amp;ref_src=twsrc%5Etfw">#faas</a> <a href="https://twitter.com/hashtag/microservices?src=hash&amp;ref_src=twsrc%5Etfw">#microservices</a> <a href="https://twitter.com/hashtag/gitops?src=hash&amp;ref_src=twsrc%5Etfw">#gitops</a> <a href="https://t.co/Af3BmXvaKl">pic.twitter.com/Af3BmXvaKl</a></p>&mdash; OpenFaaS (@openfaas) <a href="https://twitter.com/openfaas/status/1131450940807098368?ref_src=twsrc%5Etfw">May 23, 2019</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+## The `pydatascience template`
+The [pydatascience template][pydata-template] showcases a couple of ideas that anyone can leverage in their own functions and templates:
+
+1. using the [Conda package manager][conda-homepage]
+2. setting up a non-root user Python environment
+3. [multi-module Python function][multifile-blog-post]
+4. using HTTP mode in [of-watchdog][of-watchdog-homepage] to load an asset (in this case a model) into memory during startup
```

Comment:
You can extend the loading time now for the first check.. this is worth mentioning for people with large models that are loading this.